vars/documentationChecker.groovy

```diff
@@ -0,0 +1,51 @@
+
+
+// Add the following code to any Jenkinsfile.
+// It will return with success if there are no changes in the specified docs dir
+// and Will return with success if run on a non Pull Request branch
+
+// documentationChecker("docs")
+
+// docsDir ~~ the subdirectory within your repo that the documentation resides
+def documentationChecker(docsDir) {
+  if (env.CHANGE_ID != null) {
+    stage "docs PR checker"
+
+    tokens = "${env.JOB_NAME}".tokenize('/')
+    org = tokens[0]
+    repo = tokens[1]
+    branch = tokens[2]
+    sha = getOutput("git rev-parse").trim()
+    imageName = "${repo}:${branch}"
```

Comment:
2 builds of the same branch/PR but different revisions could conflict with each other without a build number or sha involved here.